src/ghci/mod.rs

```diff
@@ -323,39 +318,56 @@ impl Ghci {
         for event in events {
             let path = event.as_path();
             let path = self.relative_path(path)?;
-            // Restart on `.cabal` files, `.ghci` files, and any paths under the `restart_paths`.
-            if path.extension().map(|ext| ext == "cabal").unwrap_or(false)
+
+            let restart_match = self.opts.restart_globs.matched(&path);
+            let reload_match = self.opts.reload_globs.matched(&path);
+            let path_is_haskell_source_file = is_haskell_source_file(&path);
```

Comment:
I think we probably also want to modify the `is_haskell_source_file` impl to ignore hidden files. That'll fix the emacs bug by default.